### PR TITLE
Add esp32 example doc

### PR DESCRIPTION
**Pull request by ab-sin-the**

### CODE REVIEW

File: docs/esp32/esp32-state.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Esp32 State
+
+In this tutorial, you can monitor Wifi status (SSID and RSSI) surrounding the Esp32 device and let the device restart using Azure IoT Hub device twins.
```
Comment (Sneezry) [inline docs/esp32/esp32-state.md]: Wifi -> WiFi
Esp32 -> ESP32

File: docs/esp32/esp32-state.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+Use `F1` or `Ctrl+Shift+P` (macOS: `Cmd+Shift+P`) to open the command palette, type **IoT Workbench**, and then select **IoT Workbench: Examples**.
+
+![IoT Workbench: Examples](media/iot-workbench-examples-cmd.png)
+
+Select **ESP32 Arduino**.
+
+![IoT Workbench: Examples -> Select board](media/iot-workbench-examples-board.png)
+
+Then the **IoT Workbench Example** window is showed up.
```
Comment (Sneezry) [inline docs/esp32/esp32-state.md]: shown?

File: docs/esp32/esp32-state.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+The whole process includes:
+
+- Select an existing IoT Hub or create a new IoT Hub.
+- Select an existing IoT Hub device or create a new IoT Hub device. 
+- Create a new Function App.
+
+Please take a note of the Function App name and IoT Hub device name you created. It will be used in the next section.
+
+## Modify code for Azure Functions
+Open **esp32-state\run.csx** and modify the following line with the device name you provisioned in previous step:
```
Comment (Sneezry) [inline docs/esp32/esp32-state.md]: add an empty line after section title

File: docs/esp32/esp32-stream-analytics-cosmos-db.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+![IoT Workbench, select Esp32 ASA CosmosDB](media/esp32-stream-analytics-cosmos-db/open-example-esp32asa-cosmosdb.jpg)
+
+
+## Provision Azure Service
+
+Use `F1` or `Ctrl+Shift+P` (macOS: `Cmd+Shift+P`) to open the command palette, type **IoT Workbench**, and then select **IoT Workbench: Cloud** -> **Azure Provision**.
+
+![Azure Provision](media/iot-workbench-cloud-provision2.png)
+
+Select subscription and resource group you want to create Azure services in. A guide line show what service will be created.
```
Comment (Sneezry) [inline docs/esp32/esp32-stream-analytics-cosmos-db.md]: show -> shows

File: docs/esp32/esp32-stream-analytics-cosmos-db.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+INTO
+    "cosmosdb-2923606a-2f1c-e8c0-74e6-037933217447"
+FROM
+    "iothub-8341d26d-3fd9-3f82-aabf-3000238ff89f"
+GROUP BY
+    TumblingWindow(minute, 1)
+```
+
+
+You can design the conditions need to be satisfied to store the data into cosmosdb. In this example, only when average temperature is larger than 30 and average humidity is larger than 40 over past one minute will the data be saved.
```
Comment (Sneezry) [inline docs/esp32/esp32-stream-analytics-cosmos-db.md]: Maybe higher is better than larger (personal idea)

File: docs/esp32/m5stack-email-receiver.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+5. The designer should automatically load - if not click the **Edit** button.
+6. Select **Blank Logic App** Option to create a blank template.
+7. Search for **Gmail** and select **When a new email arrives** as trigger.
+![Gmail Trigger](media/m5stack-email-receiver/m5stack-email-receiver-gmail-trigger.png)
+8. Sign in with your Gmail account to create a connection. You can also modify the checking interval, the default value is 3 minutes.
+9. Click **New Step** and choose **Azure Functions**.
+10. Select the function you deploy in the previous step.
+11. In the **Request Body** part, enter a json format email content.
+
+		{
```
Comment (Sneezry) [inline docs/esp32/m5stack-email-receiver.md]: use code block format here

File: resources/boardlist.json [hunk 7]
```diff
@@ -28,6 +28,7 @@
             "defaultBaudRate": 115200,
             "vendorId": "10c4",
             "productId": "ea60",
+            "exampleUrl": "https://raw.githubusercontent.com/ab-sin-the/my-test-folder/master/esp32-example.json",
```
Comment (Sneezry) [inline resources/boardlist.json]: use vschina org instead

File: docs/esp32/esp32-state.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+![IoT Workbench: Examples -> Select board](media/iot-workbench-examples-board.png)
+
+Then the **IoT Workbench Example** window is shown up.
+
+![IoT Workbench, Examples window](media/iot-workbench-examples.png)
+
+Find **ESP32 State** and click **Open Sample** button. A new VS Code window with a project folder in it opens.
+
+![IoT Workbench, select ESP32 State example](media/esp32-state/open-example-esp32state.jpg)
```
Comment (lirenhe) [inline docs/esp32/esp32-state.md]: Is this pic correct?

Comment (ab-sin-the) [reply docs/esp32/esp32-state.md]: No. Tanya is working on the example images. I will update the pictures once I get those images.

**Pull request merged**